src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js

```diff
@@ -0,0 +1,172 @@
+
+import Map_ from "../../Basics/Map_/Map_";
+const _X = 1, _Y = 0;
+
+//px contains array indices [y, x]
+const getPx = (arr, px) => arr[px[_Y]][px[_X]];
+const setPx = (arr, px, val) => arr[px[_Y]][px[_X]] = val;
+
+const h_getParentIndex = i => Math.ceil(i / 2) - 1;
+const h_getChildIndex = i => i * 2 + 1;
+
+class PxHeap {
+    constructor(valueArr, indexArr) {
+        this.contents = [];
+        this.valueArr = valueArr;
+        this.indexArr = indexArr;
+    }
+
+    compare(i, j) {
+        const iVal = this.contents[i];
+        const jVal = this.contents[j];
+        return getPx(this.valueArr, iVal) > getPx(this.valueArr, jVal);
+    }
+
+    swap(i, j) {
+        const iVal = this.contents[i];
+        const jVal = this.contents[j];
+        setPx(this.indexArr, iVal, j);
+        setPx(this.indexArr, jVal, i);
+        this.contents[i] = jVal;
+        this.contents[j] = iVal;
+    }
+
+    bubbleUp(index) {
+        let parentIndex = h_getParentIndex(index);
+        while(parentIndex >= 0 && this.compare(parentIndex, index)) {
+            this.swap(index, parentIndex);
+            index = parentIndex;
+            parentIndex = h_getParentIndex(index);
+        }
+    }
+
+    bubbleDown() {
+        let index = 0, childIndex = 1;
+        let swapped = true;
+        const length = this.contents.length;
+        while(swapped && childIndex < length) {
+            swapped = false;
+            if(childIndex !== length - 1 && this.compare(childIndex, childIndex + 1))
+                childIndex++;
+            if(this.compare(index, childIndex)) {
+                this.swap(index, childIndex);
+                swapped = true;
+                index = childIndex;
+                childIndex = h_getChildIndex(index);
+            }
+        }
+    }
+
+    insert(px) {
+        let index = this.contents.length;
+        this.contents.push(px);
+        setPx(this.indexArr, px, index);
+        this.bubbleUp(index);
+    }
+
+    remove() {
+        const result = this.contents[0];
+        setPx(this.indexArr, result, -1);
+        const lastPx = this.contents.pop();
+        if(this.contents.length > 0) {
+            this.contents[0] = lastPx;
+            setPx(this.indexArr, lastPx, 0);
+            this.bubbleDown();
+        }
+        return result;
+    }
+}
+
+function pxToLatLng(px, tileBounds, zoom) {
+    const point = tileBounds.min
+        .multiplyBy(256)
+        .add([px[_X] + 0.5, px[_Y] + 0.5]);
+    return Map_.map.unproject(point, zoom);
+}
+
+//Clockwise from the right
+function moveToBacklink(move) {
+    return [
+        [ 0,  4,  0,  6,  0],
+        [ 2,  3,  5,  7,  8],
+        [ 0,  1,  0,  9,  0],
+        [16, 15, 13, 11, 10],
+        [ 0, 14,  0, 12,  0]
+    ][move[_Y] + 2][move[_X] + 2];
+}
+
+const queenMoves = [
+    [-1, -1], [-1, 0], [-1, 1],
+    [0, -1],          [0, 1],
+    [1, -1], [1, 0], [1, 1]
+];
+
+const knightMoves = [
```

Comment:
definitely explain the intent of knight moves here. the conversation we had about 8 sided vs 16 sided isochrones was really helpful